pyrsia_node/src/metadata_manager/metadata.rs

```diff
@@ -172,3 +176,702 @@ impl Iterator for PackageIterator {
         todo!()
     }
 }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// The first time that a node is run, its metadata manager needs to create the document stores that
+// it will used to hold metadata. Below are sets of definitions used to create the document stores.
+//
+// Each set of definitions has some const declarations follows by one or two function definitions.
+// The first const is for the name of the document store. The names of these follow the pattern
+// DS_documentStoreName. For example the const for the name of the document store named
+// "package_types" is DS_PACKAGE_TYPES.
+//
+// Each document store will have at least one index. Each index has a name. The pattern for const
+// names that define the name of an index is IX_documentStoreName_indexName. For example, the name
+// of the const for an index named "names" for a document store named "package_types" is
+// IX_PACKAGE_TYPES_NAMES. There will be one of these IX consts for each index that a document store
+// will have.
+//
+// Since we need to specify which fields an index will refer to, there are also const names defined
+// for each field that is covered by an index. The patter for these names is
+// FLD_documentStoreName_fieldName. For example the const name for the field named "name" in the
+// document store named "package_types" is FLD_PACKAGE_TYPES_NAME.
+//
+// For each document store to be created, there is a function named with the pattern
+// ix_documentStoreName. For example, for the document store named "package_types" the name of this
+// function is ix_package_types. This function uses the const values defined for the document store
+// to create and return a Vec of index definitions.
+//
+// For some of the document stores there is a second function named with the pattern
+// init_documentStoreName. For example, there is one of these functions for the document store named
+// "package_types". The name of the function is init_package_types. This method creates a Vec of
+// JSON strings that the document store will be pre-populated with.
+//
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Definitions for package types
+const DS_PACKAGE_TYPES: &str = "package_types";
+const IX_PACKAGE_TYPES_NAMES: &str = "names";
+const FLD_PACKAGE_TYPES_NAME: &str = "name";
```

Comment:
I would really discourage from writing documentation to explain the variable names.

https://github.com/pyrsia/pyrsia/pull/136#discussion_r772736415

```suggestion
const DOCUMENT_STORE_PACKAGE_TYPES: &str = "package_types";
const INDEX_PACKAGE_TYPES_NAMES: &str = "names";
const FIELD_PACKAGE_TYPES_NAME: &str = "name";
```

Please these should not be convoluted and hidden.